### PR TITLE
Adding the bluemix button to deploy gitlab

### PR DESCRIPTION
**Pull request by Tomcli**
Using the Bluemix Toolchain service to deploy Gitlab on your Kubernetes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,13 @@ Create a Kubernetes cluster with either [Minikube](https://kubernetes.io/docs/ge
 
 If you want to use the Bluemix Container Registry start by [Uploading the images](docs/use-bluemix-container-registry) to the Bluemix Container Registry.
 
+### Deploy to Bluemix
+If you want to deploy the wordpress directly to Bluemix, click on 'Deploy to Bluemix' button below to create a Bluemix DevOps service toolchain and pipeline for deploying the Gitlab sample, else jump to [Steps](##steps)
```
Comment (cmluciano) [inline README.md]: This sentence is to long and it has some grammar errors. 

Suggestions:
- "If you want to deploy Wordpress directly to Bluemix, click the 'Deploy to Bluemix' button below.
- The Bluemix DevOps service toolchain and pipeline is pretty wordy and should be deleted. If this is referencing a specific service, then add a hyperlink with the service name as a title.

File: scripts/bx_login.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+if [ -z $CF_ORG ]; then
+  CF_ORG="$BLUEMIX_ORG"
+fi
+if [ -z $CF_SPACE ]; then
+  CF_SPACE="$BLUEMIX_SPACE"
+fi
+
+
+if ([ -z "$BLUEMIX_USER" ] || [ -z "$BLUEMIX_PASSWORD" ] || [ -z "$BLUEMIX_ACCOUNT" ]) && ([ -z "$API_KEY"]); then
+  echo "Define all required environment variables and rerun the stage."
```
Comment (cmluciano) [inline scripts/bx_login.sh]: re-run

File: scripts/bx_login.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  exit 1
+fi
+
+echo "Deploy pods"
```
Comment (cmluciano) [inline scripts/bx_login.sh]: What is being deployed here? I'm not sure what pods refers to.

File: scripts/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  echo "Cluster $CLUSTER_NAME not created or not ready."
+  exit 1
+fi
+eval "$exp"
+
+echo -e "Deleting previous version of Gitlab if it exists"
+kubectl delete --ignore-not-found=true svc,pvc,deployment -l app=gitlab
+kubectl delete --ignore-not-found=true -f kubernetes/local-volumes.yaml
+
+kuber=$(kubectl get pods -l app=gitlab)
```
Comment (cmluciano) [inline scripts/deploy.sh]: What is the intention of this sleep?

File: scripts/install_bx.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+tar -xf /tmp/Bluemix_CLI_amd64.tar.gz --directory=/tmp
+
+# Create bx alias
+echo "#!/bin/sh" >/tmp/Bluemix_CLI/bin/bx
+echo "/tmp/Bluemix_CLI/bin/bluemix \"\$@\" " >>/tmp/Bluemix_CLI/bin/bx
+chmod +x /tmp/Bluemix_CLI/bin/*
+
+export PATH="/tmp/Bluemix_CLI/bin:$PATH"
+
+# Install Armada CS plugin
```
Comment (cmluciano) [inline scripts/install_bx.sh]: "Install the Bluemix container-service plugin"

File: scripts/install_bx.sh [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+echo "Install the Bluemix container-service plugin"
+bx plugin install container-service -r Bluemix
+
+echo "Install kubectl"
+wget --quiet --output-document=/tmp/Bluemix_CLI/bin/kubectl  https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
```
Comment (cmluciano) [inline scripts/install_bx.sh]: We should note that this demo will only work on a Linux box or add a bit more checks to pull for OSX if detected.

File: scripts/deploy.sh [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+fi
+
+echo -e "Creating pods"
+kubectl create -f kubernetes/local-volumes.yaml
+kubectl create -f kubernetes/postgres.yaml
+sleep 5s
+kubectl create -f kubernetes/redis.yaml
+sleep 5s
+kubectl create -f kubernetes/gitlab.yaml
+kubectl get nodes
```
Comment (cmluciano) [inline scripts/deploy.sh]: What is the ```get nodes``` check for?

File: scripts/deploy.sh [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+kuber=$(kubectl get pods -l app=gitlab)
+if [ ${#kuber} -ne 0 ]; then
+	sleep 30s
+fi
+
+echo -e "Creating pods"
+kubectl create -f kubernetes/local-volumes.yaml
+kubectl create -f kubernetes/postgres.yaml
+sleep 5s
```
Comment (cmluciano) [inline scripts/deploy.sh]: These sleeps are not necessary. Kubernetes can handle multiple creates.

File: scripts/install_bx.sh (hunk 6 above)
Comment (Tomcli) [inline scripts/install_bx.sh]: We are deploying Gitlab using the [Bluemix Toolchain](https://console.ng.bluemix.net/docs/services/ContinuousDelivery/toolchains_about.html#toolchains_about) where it only runs on Linux.

**Comment by animeshsingh** [comment]
LGTM

**Pull request merged**